src/clientiface.h

```diff
@@ -201,10 +230,24 @@ class RemoteClient
 	//
 	u16 net_proto_version;
 
+	/* Authentication information */
+	std::string enc_pwd;
+	int chosen_mech;
+	void * auth_data;
+	u32 allowed_auth_mechs;
+	u32 allowed_sudo_mechs;
+
+	bool is_sudo_mech_allowed(int mech)
```

Comment:
Pedantically, this would preferably be `bool is_sudo_mech_allowed(unsigned mech)`